iib/web/messaging.py

```diff
@@ -59,6 +63,56 @@ def __init__(
             if failed and self.conn:
                 self.close()
 
+    def wait(self, condition, timeout=False, msg=None):
+        """
+        Process events until ``condition()`` returns ``True``.
+
+        :param condition: Condition which determines when the wait will end.
+        :type condition: Function which returns ``bool``
+        :param timeout: Timeout in seconds. If ``False``, the value of ``timeout`` used in the
+            constructor of this object will be used. If ``None``, there is no timeout. Any other
+            value is treated as a timeout in seconds.
+        :type timeout: ``None``, ``False``, ``float``
+        :param msg: Context message for :class:`proton.Timeout` exception
+        :type msg: ``str``
+        """
+        if timeout is False:
+            timeout = self.timeout
+        if timeout is None:
+            while not condition() and not self.disconnected:
+                self.container.process()
+        else:
+            container_timeout = self.container.timeout
+            self.container.timeout = timeout
```

Comment:
Should we check for the type of ‘timeout’ ? What if someone passes in a string for instance?

Reply:
@yashvardhannanavati I didn't write this part of the code. The only code I added was https://github.com/apache/qpid-proton/pull/243.

I'm trying to keep the code as close to upstream as possible.

Reply:
+1 for minimizing the changes.

Reply:
Agreed! But we are anyways never going to be exactly at par with the upstream code. FWIW, I feel it might be okay to add a couple of lines. But if Luiz and you feel differently, I am happy to let it go.

Reply:
@yashvardhannanavati the idea is that if the PR upstream gets merged, then we would just use the library and make any customizations in IIB.